Scripts/UpdateCommandNames.py

```diff
@@ -0,0 +1,148 @@
+import os
+import json
+from datetime import datetime
+import shutil
+
+ParentPath = os.path.dirname(os.path.dirname(__file__))
+
+# Set the path where you want to save this new Json file
+# JsonPath = os.path.dirname(__file__)
+JsonPath = ParentPath
+
+# Set the file name. Default is "RibbonStructure.json".
+# This is the file used to reset the ribbon.
+JsonName = "RibbonStructure.json"
+
+# Define list of the workbenches, toolbars and commands on class level
+List_Workbenches = []
+Dict_Commands = {}
+Dict_RibbonCommandPanel = {}
+
+
+def main():
+    ReadJson()
+    ReadCommands()
+    UpdateCommands()
+    WriteJson()
+
+
+def ReadJson():
+    """Read the Json file and fill the lists and set settings"""
+    # Open the JsonFile and load the data
+    JsonFile = open(os.path.join(JsonPath, "RibbonStructure.json"))
+    data = json.load(JsonFile)
+
+    # Create a backup file
+    OriginalPath = os.path.join(JsonPath, "RibbonStructure.json")
+    Suffix = datetime.now().strftime("%Y%m%d_%H%M%S")
+    BackupName = f"RibbonStructure_{Suffix}.json"
+    pathBackup = os.path.dirname(__file__) + "/Backups"
+    if os.path.exists(pathBackup) is False:
+        os.makedirs(pathBackup)
+    BackupFile = os.path.join(pathBackup, BackupName)
+    shutil.copy(OriginalPath, BackupFile)
+
+    # Get the dict with the customized date for the buttons
+    try:
+        Dict_RibbonCommandPanel["workbenches"] = data["workbenches"]
+    except Exception as e:
+        print(e)
+        pass
+
+    JsonFile.close()
+    return
+
+
+def ReadCommands():
+    """Read the Json file and fill the lists and set settings"""
+    # Open the JsonFile and load the data
+    JsonFile = open(os.path.join(JsonPath, "CommandList.json"))
+    data = json.load(JsonFile)
+
+    # Get the dict with the customized date for the buttons
+    try:
+        Dict_Commands.update(data)
+    except Exception as e:
+        print(e)
+        pass
+
+    JsonFile.close()
+    return
+
+
+def add_keys_nested_dict(dict, keys):
+    for key in keys:
+        if key not in dict:
+            dict[key] = {}
+        dict = dict[key]
+    try:
+        dict.setdefault(keys[-1], 1)
+    except Exception:
+        pass
+    return
+
+
+def UpdateCommands():
+    for WorkBench in Dict_RibbonCommandPanel["workbenches"]:
+        for ToolBar in Dict_RibbonCommandPanel["workbenches"][WorkBench]["toolbars"]:
+            if ToolBar != "order":
+                for Command in Dict_RibbonCommandPanel["workbenches"][WorkBench][
+                    "toolbars"
+                ][ToolBar]["commands"]:
+                    for key, value in Dict_Commands.items():
+                        if Command == key:
+                            if value[2] != "" and value[2] != "...":
+                                Dict_RibbonCommandPanel["workbenches"][WorkBench][
+                                    "toolbars"
+                                ][ToolBar]["commands"][Command]["text"] = value[2]
```

Comment:
This function has four nested for-loops plus three nested if clauses. That is quite a lot :D
Please break this into smaller functions or at least add a few comments.

Reply:
I've added some comments to clarify